solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityHandler.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2020-2020. AxonIQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the &quot;License&quot;)
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.axoniq.demo.hotel.booking.query;
+
+import org.axonframework.config.ProcessingGroup;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.queryhandling.QueryHandler;
+import org.axonframework.queryhandling.QueryUpdateEmitter;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Component
+@ProcessingGroup("room-availability")
+class RoomAvailabilityHandler {
+
+    private final RoomAvailabilityEntityRepository roomAvailabilityEntityRepository;
+    private final QueryUpdateEmitter queryUpdateEmitter;
+
+    private RoomAvailabilityResponseData convert(RoomAvailabilityEntity entity, UUID accountId) {
+        return new RoomAvailabilityResponseData(entity.getRoomNumber(),
+                                                entity.getRoomDescription(),
+                                                entity.getRoomStatus(),
+                                                accountId==null ? entity.getBookings().stream().map(it -> new BookingData(it.getId(), it.getStartDate(), it.getEndDate())).collect(Collectors.toUnmodifiableList()) : entity.getBookings().stream().filter(it -> it.getAccountId().equals(accountId.toString())).map(it -> new BookingData(it.getId(), it.getStartDate(), it.getEndDate())).collect(Collectors.toUnmodifiableList()),
```

Comment:
This line is 422 characters wide, which is a bit much. Can we format this and use variables?

Reply:
This is something we should fix in master